### PR TITLE
Remove bower

### PR DESCRIPTION
**Pull request by gyss**
Upgraded origami component versions, removed Bower, and publish the package to NPM when a new tag is created

Origami dependencies updated. All of them (except `o-overlay`) the only breaking change is the Bower deprecation:

- o-buttons: [6.0.2 -> 7.3.0](https://github.com/Financial-Times/origami/blob/main/components/o-buttons/MIGRATION.md#migrating-from-v6-to-v7)
- o-colors: [5.4.1 -> 6.4.2](https://github.com/Financial-Times/origami/blob/main/components/o-colors/MIGRATION.md#migrating-from-v5-to-v6)
- o-icons: ^[6.3.0 -> 7.2.1](https://github.com/Financial-Times/origami/blob/main/components/o-icons/MIGRATION.md#migrating-from-v6-to-v7)
- o-message: ^[4.1.8 -> 5.2.1](https://github.com/Financial-Times/origami/blob/main/components/o-message/MIGRATION.md#migrating-from-v4-to-v5)
- o-overlay: [^3.0.0 -> 4.2.3](https://github.com/Financial-Times/origami/blob/main/components/o-overlay/MIGRATION.md#migrating-from-v3-to-v4) - deprecated events not used
- o-tracking: [^3.0.0 -> 4.2.1](https://github.com/Financial-Times/origami/blob/main/libraries/o-tracking/MIGRATION.md#migrating-from-v3-to-v4)
- o-viewport: ^5.0.0 - no updates
- @financial-times/o-viewport: [^4.0.0 -> 5.1.1](https://github.com/Financial-Times/origami/blob/main/components/o-viewport/MIGRATION.md#migrating-from-v4-to-v5)
- o-visual-effects: [^3.0.0 -> 4.1.1](https://github.com/Financial-Times/origami/tree/main/components/o-visual-effects) 

Tested in local in `next-home-page` using npm link and hardcoding the user maintenance mode.

Screenshot 2022-02-14 at 15 26 42

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -104,6 +104,18 @@ jobs:
           path: test-results
           destination: test-results
 
+  publish:
```
Comment (asugar13) [inline .circleci/config.yml:107]: **Question**: why is this new step required?

File: .circleci/config.yml [hunk 2]
```diff
@@ -117,6 +129,22 @@ workflows:
           requires:
             - build
 
+  build-test-publish:
```
Comment (asugar13) [inline .circleci/config.yml:132]: **Question**: Why is this new CI stage needed?

File: jest-resolver.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+module.exports = (request, options) => {
```
Comment (asugar13) [inline jest-resolver.js:1]: **Question**: I haven't seen this in the guide to migrate origami components. Is this needed on all migrations from bower to npm?

File: .circleci/config.yml (hunk 1 above)
Comment (gyss) [reply .circleci/config.yml:107]: this step is meant to publish the `n-syndication` package on npm

File: .circleci/config.yml (hunk 2 above)
Comment (gyss) [reply .circleci/config.yml:132]: this workflow will be triggered when a new tag is published, and this will run the jobs required to publish on npm

**Review by asugar13 — COMMENTED** [review]
looks good. just left a few questions

File: jest-resolver.js (hunk 3 above)
Comment (gyss) [reply jest-resolver.js:1]: So I had an issue when running the Jest tests with the new updated components. Specifically, the `o-overlay` component has published only the Browserify entry point (`"browser": '"main.js" [here](https://github.com/Financial-Times/origami/blob/8e60e8210c871b9adaf4f09a4b92f1a49ef4ce56/components/o-overlay/package.json#L21) and Jest can't import properly. In order to fix this found a solution by writing a Jest resolver that will include the `browser` field from all packages.

Comment (asugar13) [reply jest-resolver.js:1]: thank you. I will keep an eye out for future repos.

**Pull request merged**